### PR TITLE
feat(webdriver): support `Page.deleteCookie()` for WebDriver BiDi

### PR DESCRIPTION
**Pull request by sadym-chromium**
Implement `Page.deleteCookie()` via BiDi `storage.deleteCookies`.

### CODE REVIEW

**Comment by OrKoN** [comment]
> Should we extend tests verifying this?

Let's add a test to also understand the cdp behavior better.

File: packages/puppeteer-core/src/bidi/core/UserContext.ts [hunk 1]
```diff
@@ -196,6 +197,26 @@ export class UserContext extends EventEmitter<{
...
+  })
+  async deleteCookie(...cookies: DeleteCookiesRequest[]): Promise<void> {
+    await Promise.all(cookies.map(async deleteCookieRequest => {
+      const filter = {
+        ...deleteCookieRequest
+      };
+      await this.#session.send('storage.deleteCookies', {
+        filter: filter,
+        partition: {
+          type: 'storageKey',
```
Comment (OrKoN) [inline packages/puppeteer-core/src/bidi/core/UserContext.ts]: should we delete using the type=context instead?

Comment (OrKoN) [reply packages/puppeteer-core/src/bidi/core/UserContext.ts]: moving it to the BrowseringContext.ts correspondingly

**Review by OrKoN — COMMENTED** [review]
Probably need to enable some tests via TestExpectations

File: packages/puppeteer-core/src/bidi/core/UserContext.ts (hunk 1 above)
Comment (sadym-chromium) [reply packages/puppeteer-core/src/bidi/core/UserContext.ts]: This depends on the desirable observable differences between firefox and chrome. Let me add a test for it

File: packages/puppeteer-core/src/bidi/Page.ts [hunk 2]
```diff
@@ -559,9 +560,6 @@ export class BidiPage extends Page {
         ),
       };
 
-      // TODO: delete cookie before setting them.
-      // await this.deleteCookie(bidiCookie);
-
```
Comment (sadym-chromium) [inline packages/puppeteer-core/src/bidi/Page.ts:564]: The scope of set and deleted cookie can be different, so deleting can cause some unwilling side effects. Example: a cookie is set for a specific path, but removed from the whole domain.

File: test/src/cookies.spec.ts [hunk 3]
```diff
@@ -584,5 +584,139 @@ describe('Cookie specs', () => {
...
+          size: 33,
+          httpOnly: false,
+          secure: false,
+          session: true,
+          sourceScheme: 'NonSecure',
+        },
+      ]);
+    });
+    it('should delete cookie for specified URL regardless of the current page', async () => {
+      // Tentative test case. Depending on the cookie partitioning implementation, this
```
Comment (sadym-chromium) [inline test/src/cookies.spec.ts]: This is an edge case where firefox and chrome can have observable different behavior. In ff all the cookies are partitioned, while in chrome they are not. In ff if context partition is used, delete command scoped to the cookies set by the current page. In chrome it is scoped only for the user context. As discussed offline, this is an expected to have different behavior for different implementations.

File: packages/puppeteer-core/src/bidi/core/UserContext.ts (hunk 1 above)
Comment (sadym-chromium) [reply packages/puppeteer-core/src/bidi/core/UserContext.ts]: Switched to `BrowsingContextPartitionDescriptor` and added tests.

File: test/TestExpectations.json [hunk 4]
```diff
@@ -1080,6 +1080,54 @@
     "parameters": ["cdp", "firefox"],
     "expectations": ["FAIL"]
   },
+  {
```
Comment (OrKoN) [inline test/TestExpectations.json:1090]: Could you please add a `"comment":""` field with a comment about the reason for the fail expectation?

File: test/TestExpectations.json [hunk 5]
```diff
@@ -1080,6 +1080,54 @@
...
+    "testIdPattern": "[cookies.spec] Cookie specs Page.deleteCookie should delete cookie",
+    "platforms": ["linux"],
```
Comment (OrKoN) [inline test/TestExpectations.json]:
```suggestion
    "platforms": ["linux"],
```

all platforms?

File: test/TestExpectations.json [hunk 6]
```diff
@@ -1080,6 +1080,54 @@
...
+    "testIdPattern": "[cookies.spec] Cookie specs Page.deleteCookie should delete cookie for specified URL",
+    "platforms": ["linux"],
+    "parameters": ["cdp", "firefox"],
+    "expectations": ["FAIL"]
+  },
+  {
+    "testIdPattern": "[cookies.spec] Cookie specs Page.deleteCookie should delete cookie for specified URL regardless of the current page",
+    "platforms": ["linux"],
+    "parameters": ["firefox", "webDriverBiDi"],
+    "expectations": ["SKIP"]
```
Comment (OrKoN) [inline test/TestExpectations.json]: I assume the skipped once are the ones that are due to different implementation in partitioning? (let's leave a comment)

File: test/TestExpectations.json (hunk 4 above)
Comment (sadym-chromium) [reply test/TestExpectations.json:1090]: done

File: test/TestExpectations.json (hunk 5 above)
Comment (sadym-chromium) [reply test/TestExpectations.json]: indeed, done

File: test/TestExpectations.json (hunk 6 above)
Comment (sadym-chromium) [reply test/TestExpectations.json]: done

File: test/TestExpectations.json (hunk 4 above)
Comment (OrKoN) [reply test/TestExpectations.json:1090]: Sorry, I was not clear on this. I meant to ask for a comment for each of the expectations that are new

**Review by OrKoN — APPROVED** [review]
